sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +379,482 @@ def transform(self, X):
+            The imputed input data.
+        """
+        if self.estimator is None:
+            from ..linear_model import BayesianRidge
+            self.estimator_ = BayesianRidge()
+        else:
+            self.estimator_ = clone(self.estimator)
+
+        # parse min and max values
+        self.min_value_ = np.nan if self.min_value is None else self.min_value
+        self.max_value_ = np.nan if self.max_value is None else self.max_value
+
+        # check that the estimator's predict method has return_std argument
+        if 'return_std' not in signature(self.estimator_.predict).parameters:
+            raise ValueError(
+                "The regression estimator is %s and its predict method does "
+                "not support 'return_std=True'. This is required for MICE." %
+                type(self.estimator_)
+            )
+
+        # initial imputation
+        self.initial_imputer_ = None
+        X, X_filled, mask_missing_values = self._initial_imputation(X)
+        n_samples, n_features = X.shape
+
+        # perform imputations
+        n_rounds = self.n_burn_in + self.n_imputations
+        results_list = []
+        self.trained_estimator_triplets_ = []
+        if self.verbose:
+            print("[MICE] Completing matrix with shape %s" % (X.shape,))
+            start_t = time()
+        for round in range(n_rounds):
+            # order in which to impute
+            ordered_idx = self._get_ordered_idx(mask_missing_values)
+
+            # abs_correlation matrix is used to choose a subset of other
+            # features to impute from
+            abs_corr_mat = self._get_abs_correlation_matrix(X_filled)
+
+            # Fill in each feature in the order of ordered_idx
+            for feat_idx in ordered_idx:
+                neighbor_feat_idx = self._get_neighbor_feat_idx(n_features,
+                                                                feat_idx,
+                                                                abs_corr_mat)
+                X_filled, estimator = self._impute_one_feature(
+                    X_filled, mask_missing_values, feat_idx, neighbor_feat_idx,
+                    estimator=None)
+                estimator_triplet = (feat_idx, neighbor_feat_idx, estimator)
+                self.trained_estimator_triplets_.append(estimator_triplet)
+
+            if round >= self.n_burn_in:
+                results_list.append(X_filled[mask_missing_values])
+            if self.verbose:
+                print('[MICE] Ending imputation round round '
+                      '%d/%d, elapsed time %0.2f'
+                      % (round + 1, n_rounds, time() - start_t))
+
+        if len(results_list) > 0:
+            X[mask_missing_values] = np.array(results_list).mean(axis=0)
+        else:
+            X[mask_missing_values] = X_filled[mask_missing_values]
+
+        return X
+
+    def transform(self, X):
+        """Imputes all missing values in X.
+
```

Comment:
Note that this is stochastic, and that if random_state is not fixed, repeated calls, or permuted input, will yield different results.

Reply:
So noted.